go-controller/pkg/libovsdbops/portgroup.go

```diff
@@ -21,6 +21,28 @@ func FindPortGroupsWithPredicate(nbClient libovsdbclient.Client, p portGroupPred
 	return found, err
 }
 
+// GetPortGroupACLsPred returns ACL predicate to filter objects referenced by given port groups.
+// aclPred can be additionally applied to the returned predicate, nil will be ignored
+func GetPortGroupACLsPred(portGroups []*nbdb.PortGroup, aclPred func(item *nbdb.ACL) bool) func(item *nbdb.ACL) bool {
+	aclUUIDs := map[string]bool{}
+	for _, pg := range portGroups {
+		for _, aclUUID := range pg.ACLs {
+			aclUUIDs[aclUUID] = true
+		}
+	}
+	var pgACLPred func(item *nbdb.ACL) bool
+	if aclPred == nil {
+		pgACLPred = func(item *nbdb.ACL) bool {
+			return aclUUIDs[item.UUID]
+		}
+	} else {
+		pgACLPred = func(item *nbdb.ACL) bool {
+			return aclUUIDs[item.UUID] && aclPred(item)
+		}
+	}
+	return pgACLPred
+}
+
```

Comment:
I don't think this should be in libovsdbops. Probably better in policy.go if it is used only there, or maybe in the new acl.go.